src/fobnail-attester.c

```diff
@@ -129,6 +130,78 @@ static void coap_aik_handler(struct coap_resource_t* resource, struct coap_sessi
 
 }
 
+UsefulBuf encode_meta(struct meta_data *meta)
+{
+    UsefulBuf Buffer = NULLUsefulBuf;
+    QCBOREncodeContext ctx;
+    QCBOREncode_Init(&ctx, Buffer);
+
+    QCBOREncode_OpenMap(&ctx);
+        QCBOREncode_AddUInt64ToMap(&ctx, "version", meta->header_version);
+        QCBOREncode_OpenArrayInMap(&ctx, "mac");
+        for (int i = 0; i < 6; i++)
+            QCBOREncode_AddInt64(&ctx, meta->mac[i]);
+
+        QCBOREncode_CloseArray(&ctx);
+        QCBOREncode_AddSZStringToMap(&ctx, "manufacturer", meta->manufacturer);
+        QCBOREncode_AddSZStringToMap(&ctx, "product_name", meta->product_name);
+        QCBOREncode_AddSZStringToMap(&ctx, "serial_number", meta->serial_number);
+
+    QCBOREncode_CloseMap(&ctx);
+
+    UsefulBufC EncodedCBOR;
+    QCBORError uErr;
+    uErr = QCBOREncode_Finish(&ctx, &EncodedCBOR);
```

Comment:
After this line `meta->manufacturer`, `meta->product_name` and `meta->serial_number` can and should be freed.

Reply:
@krystian-hebel  Indeed. Thanks. Fixed.